src/GraphQL/Types/Scalars/DateGraphType.cs

```diff
@@ -19,47 +19,52 @@ public DateGraphType()
         }
 
         /// <inheritdoc/>
-        public override object Serialize(object value)
+        public override object ParseLiteral(IValue value) => value switch
         {
-            var date = ParseValue(value);
-
-            if (date is DateTime dateTime)
-            {
-                return dateTime.ToString("yyyy-MM-dd", CultureInfo.InvariantCulture);
-            }
-
-            return null;
-        }
+            NullValue _ => null,
+            StringValue stringValue => ParseDate(stringValue.Value),
+            _ => ThrowLiteralConversionError(value)
+        };
 
         /// <inheritdoc/>
-        public override object ParseLiteral(IValue value)
-            => value is StringValue stringValue ? ParseValue(stringValue.Value) : null;
+        public override object ParseValue(object value) => value switch
+        {
+            DateTime d => ValidateDate(d, value), // no boxing
+            string stringValue => ParseDate(stringValue),
+            null => null,
+            _ => throw new FormatException($"Could not parse date. Expected either a string or a DateTime without time component. Value: {value}")
```

Comment:
Is this the only scalar that generates `FormatException` now?

Reply:
Probably.  Should we change it?  I like the detailed error message.

Reply:
```suggestion
            _ => throw new InvalidOperationException($"Could not parse date. Expected either a string or a DateTime without time component. Value: {value}")
```
Tests may need to change also.

Reply:
Basically it's legal to throw any exception from scalar's method. I see many `OverflowException` for example. We may keep here `FormatException` as is. But remember that exception then will be passed to response writer which calculates error code - `FORMAT`/`INVALID_OPERATION`/`OVERFLOW`. In some cases, it is beneficial to have fewer codes for the caller, in some more. I have no strict opinion here.